cppcache/integration/test/RegionPutTest.cpp

```diff
@@ -0,0 +1,294 @@
+  auto pool = createPool(cluster, cache);
+  auto region = setupRegion(cache, pool);
+  auto key = CacheableKey::create("key-1");
+
+  EXPECT_NO_THROW(region->put(key, CacheableString::create("previous-value")));
+
+  auto value = std::dynamic_pointer_cast<CacheableString>(region->get(key));
+
+  auto previous =
+      region->putIfAbsent(key, CacheableString::create("new-value"));
+  EXPECT_THAT(previous, NotNull());
+
+  auto converted = std::dynamic_pointer_cast<CacheableString>(previous);
+  EXPECT_THAT(converted, NotNull());
+  EXPECT_THAT(converted->toString(), Eq(value->toString()));
+}
+
+TEST(RegionPutTest, testPutIfAbsentExistingEntryCaching) {
+  Cluster cluster{LocatorCount{1}, ServerCount{1}};
+
+  cluster.start();
+
+  cluster.getGfsh()
+      .create()
+      .region()
+      .withName("region")
+      .withType("REPLICATE")
+      .execute();
+
+  auto cache = createCache();
+  auto pool = createPool(cluster, cache);
+  auto region = setupCachingRegion(cache, pool);
+  auto key = CacheableKey::create("key-1");
+
+  EXPECT_NO_THROW(region->put(key, CacheableString::create("previous-value")));
+
+  auto value = std::dynamic_pointer_cast<CacheableString>(region->get(key));
+
+  auto previous =
+      region->putIfAbsent(key, CacheableString::create("new-value"));
+  EXPECT_THAT(previous, NotNull());
+
+  auto converted = std::dynamic_pointer_cast<CacheableString>(previous);
+  EXPECT_THAT(converted, NotNull());
+  EXPECT_THAT(converted->toString(), Eq(value->toString()));
+
+  // Verify cached value matches expected
+  auto entry = region->getEntry(key);
+  EXPECT_THAT(entry, NotNull());
+
+  auto retrieved = entry->getValue();
+  EXPECT_THAT(converted, NotNull());
+
+  converted = std::dynamic_pointer_cast<CacheableString>(retrieved);
+  EXPECT_THAT(converted, NotNull());
+  EXPECT_THAT(converted->toString(), Eq(value->toString()));
+}
+
+TEST(RegionPutTest, testPutIfAbsentWIthinTxAndOutOfTx) {
+  Cluster cluster{LocatorCount{1}, ServerCount{1}};
+
+  cluster.start();
+
+  cluster.getGfsh()
+      .create()
+      .region()
+      .withName("region")
+      .withType("REPLICATE")
+      .execute();
+
+  auto cache = createCache();
+  auto pool = createPool(cluster, cache);
+  auto region = setupRegion(cache, pool);
+  auto key = CacheableKey::create("key-1");
+  auto value = CacheableString::create("value");
+  auto txMgr = cache.getCacheTransactionManager();
+
+  txMgr->begin();
+  EXPECT_THAT(region->putIfAbsent(key, value), IsNull());
+
+  auto retrieved = region->get(key);
+  EXPECT_THAT(retrieved, NotNull());
+
+  auto converted = std::dynamic_pointer_cast<CacheableString>(retrieved);
+  EXPECT_THAT(converted, NotNull());
+  EXPECT_THAT(converted->toString(), Eq(value->toString()));
+  auto& txId = txMgr->suspend();
+
+  EXPECT_THAT(region->putIfAbsent(key, value), IsNull());
+
+  txMgr->resume(txId);
+  EXPECT_THROW(txMgr->commit(), CommitConflictException);
+}
+
+TEST(RegionPutTest, testPutIfAbsentExistingEntryTX) {
```

Comment:
What is the difference between this test case and `testPutIfAbsentExistingEntryCaching()`?
Why is there a TX suffix in the name?